activerecord/lib/active_record/connection_adapters/abstract/quoting.rb

```diff
@@ -17,7 +22,11 @@ def quote(value)
       # SQLite does not understand dates, so this method will convert a Date
       # to a String.
       def type_cast(value, column = nil)
+        value = id_value_for_database(value) if value.is_a?(Base)
+
         if value.respond_to?(:quoted_id) && value.respond_to?(:id)
+          ActiveSupport::Deprecation.warn \
```

Comment:
We don't need to deprecate here, it is only being used to make sure it is a AR::Base object.